src/algebra/algebra/basic.lean

```diff
@@ -1404,20 +1404,49 @@ end linear_map
 end is_scalar_tower
 
 section restrict_scalars
-/- In this section, we describe restriction of scalars: if `S` is an algebra over `R`, then
-`S`-modules are also `R`-modules. -/
 
 section type_synonym
 variables (R A M : Type*)
 
-/--
-Warning: use this type synonym judiciously!
-The preferred way of working with an `A`-module `M` as `R`-module (where `A` is an `R`-algebra),
-is by `[module R M] [module A M] [is_scalar_tower R A M]`.
+/-- A type synonym representing the functor from the category of modules over `A`, regarded as a
+(semi)ring, to the category of modules over `A`, regarded as an `R`-algebra.
 
-When `M` is a module over a ring `A`, and `A` is an algebra over `R`, then `M` inherits a
-module structure over `R`, provided as a type synonym `module.restrict_scalars R A M := M`.
--/
+Warning: use this type synonym judiciously! Consider an example where we want to construct an
+`R`-linear map from `M` to `A`, given:
+```lean
+variables (R A M : Type*)
+variables [comm_semiring R] [semiring A] [algebra R A] [add_comm_monoid M] [module A M]
+```
+With the assumptions above we can't directly state our map as we have no `module R M` structure, but
+`restrict_scalars` permits it to be written as:
+```lean
+-- an `R`-module structure on `M` is provided by `restrict_scalars` which is compatible
+example : restrict_scalars R A M →ₗ[R] A := sorry
+```
+However, it is usually better just to add this extra structure as an argument:
+```lean
+-- an `R`-module structure on `M` and proof of its compatibility is provided by the user
+example [module R M] [is_scalar_tower R A M] : M →ₗ[R] A := sorry
+```
+The advantage of the second approach is that it defers the duty of providing the missing typeclasses
+`[module R M] [is_scalar_tower R A M]`. If some concrete `M` naturally carries these (as is often
+the case) then we have avoided `restrict_scalars` entirely. If not, we can pass
+`restrict_scalars R A M` later on instead of `M`.
+
+Note that this means we almost always want to state definitions and lemmas in the language of
+`is_scalar_tower` rather than `restrict_scalars`.
+
+Mathematically, when `A` is an `R`-algebra, there are two different functors:
```

Comment:
I am completely confused by this remark. I thought that `restrict_scalars` is exactly meant to be used for (2) in situations where we need a "construction" (as opposed to just adding an `is_scalar_tower` assumption.

I don't understand what (1) means in Lean anyway. We only have modules over semirings. I don't think Lean distinguishes between modules over the semiring `A` and modules over the `R`-algebra `A`.

Reply:
I've just removed this paragraph as I think it's unlikely to help no matter what the form.

Reply:
@ocfnash Aha! Thanks a lot for your explanation. I agree that removing it is ok. But if you want to restore it in some form, then this last version with informal mathematical language is certainly the one that was most helpful to me.

Reply:
I'm now convinced removing the paragraph is the way to go. The quotient of the effort required to actually make this extra point clearly versus its actual value is extremely low!

I think the job of this comment is to say to the person: "are you sure you shouldn't be using `is_scalar_tower` instead?" and it does that.